packages/api-client-core/spec/operationBuilders.spec.ts

```diff
@@ -667,6 +667,119 @@ describe("operation builders", () => {
       `);
     });
 
+    test("actionOperation should build a mutation query for an object hasReturnType", async () => {
+      expect(
+        actionOperation(
+          "upsertWidget",
+          { __typename: true, id: true, name: true, eventAt: true },
+          "widget",
+          "widget",
+          {},
+          null,
+          null,
+          null,
+          {
+            "... on CreatePostResult": { hasReturnType: false },
+            "... on UpdatePostResult": { hasReturnType: false },
+          }
+        )
+      ).toMatchInlineSnapshot(`
+        {
+          "query": "mutation upsertWidget {
+          upsertWidget {
+            success
+            errors {
+              message
+              code
+              ... on InvalidRecordError {
+                validationErrors {
+                  message
+                  apiIdentifier
+                }
+              }
+            }
+            ... on CreatePostResult {
+              __typename
+              widget {
+                __typename
+                id
+                name
+                eventAt
+              }
+            }
+            ... on UpdatePostResult {
+              __typename
+              widget {
+                __typename
+                id
+                name
+                eventAt
+              }
+            }
+          }
+          gadgetMeta {
+            hydrations(modelName: "widget")
+          }
+        }",
+          "variables": {},
+        }
+      `);
+    });
+
+    test("actionOperation should build a mutation query for an object hasReturnType with an inner return type", async () => {
+      expect(
+        actionOperation(
+          "upsertWidget",
+          { __typename: true, id: true, name: true, eventAt: true },
+          "widget",
+          "widget",
+          {},
+          null,
+          null,
+          null,
+          {
+            "... on CreatePostResult": { hasReturnType: false },
+            "... on UpdatePostResult": { hasReturnType: true },
+          }
+        )
+      ).toMatchInlineSnapshot(`
+        {
+          "query": "mutation upsertWidget {
+          upsertWidget {
+            success
+            errors {
+              message
+              code
+              ... on InvalidRecordError {
+                validationErrors {
+                  message
+                  apiIdentifier
+                }
+              }
+            }
+            ... on CreatePostResult {
```

Comment:
since these are the type of the action return type now they would change when the action api identifier changes;

I'll make sure to test this on the api side PR